### PR TITLE
BibFormat: new enhanced MARCXML format

### PR DESCRIPTION
**Pull request by kaplun**
* Introduces new XME format with enhanced MARCXML sporting:
         100/700 $x with Record ID of linked HepName,
                 $y with True/False if the signature is claimed
                 $z with Record ID of institution
         371/110 $z with Record ID of institution
         502     $z with Record ID of institution
         999C5   $0 with on the fly discovered Record IDs (not for books)
         773     $0 with Record ID of corresponding Journal or Conference
         693/710 $0 with Record ID of corresponding experiment

* Introduces new XOAIMARCE corresponding format for OAI-PMH.

Signed-off-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

File: bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+    Adds 100/700 $x with Record ID of linked HepName,
+                 $y with True/False if the signature is claimed
+                 $z with Record ID of institution
+         371/110 $z with Record ID of institution
+         502     $z with Record ID of institution
+         999C5   $0 with on the fly discovered Record IDs (not for books)
+         773     $0 with Record ID of corresponding Journal or Conference
+         693/710 $0 with Record ID of corresponding experiment
+    """
```
Comment (kaplun) [inline bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py]: @annetteholtkamp, @tsgit, @ksachs what do you think? Do you know if there are other hidden links that can be made more explicit?

File: bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    record = bfo.get_record()
+    recid = bfo.recID
+
+    # Let's add signatures
+    for field in record_get_field_instances(record, '100') + record_get_field_instances(record, '700'):
+        subfields = field_get_subfield_instances(field)
+        subfield_dict = dict(subfields)
+        if 'a' in subfield_dict:
+            author_name = subfield_dict['a']
+            rows = run_sql("SELECT personid, flag FROM aidPERSONIDPAPERS WHERE bibrec=%s AND name=%s AND flag>-2", (recid, author_name))
```
Comment (jalavik) [inline bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py]: Do we have indices in the database tables for these WHERE selectors? If not, it should be added where appropriate.

Comment (kaplun) [inline bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py]: Yep:
```sql
MariaDB [inspirehep]> explain SELECT personid, flag FROM aidPERSONIDPAPERS WHERE bibrec=123 AND name="Storaci, B" AND flag>-2;
+------+-------------+-------------------+------+---------------------+-------+---------+-------+------+-------------+
| id   | select_type | table             | type | possible_keys       | key   | key_len | ref   | rows | Extra       |
+------+-------------+-------------------+------+---------------------+-------+---------+-------+------+-------------+
|    1 | SIMPLE      | aidPERSONIDPAPERS | ref  | rec-b,name-b,flag-b | rec-b | 3       | const |    1 | Using where |
+------+-------------+-------------------+------+---------------------+-------+---------+-------+------+-------------+
1 row in set (0.00 sec)
```

Comment (jalavik) [inline bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py]: Cool!

**Pull request merged**